src/ai/backend/manager/models/scaling_group.py

```diff
@@ -315,6 +320,42 @@ class ScalingGroup(graphene.ObjectType):
     scheduler_opts = graphene.JSONString()
     use_host_network = graphene.Boolean()
 
+    # Dynamic fields.
+    agent_count = graphene.Int(description="Added in 24.03.5.")
+    agent_total_resource_slots = graphene.JSONString(description="Added in 24.03.5.")
+
+    async def resolve_agent_count(self, info: graphene.ResolveInfo) -> int:
+        from .agent import Agent
+
+        return await Agent.load_count(
+            info.context,
+            scaling_group=self.name,
+        )
+
+    async def resolve_agent_total_resource_slots(
+        self, info: graphene.ResolveInfo
+    ) -> Mapping[str, Any]:
+        from .agent import AgentRow
+
+        graph_ctx = info.context
+        async with graph_ctx.db.begin_readonly() as db_session:
+            query_stmt = sa.select([AgentRow.occupied_slots, AgentRow.available_slots]).where(
+                AgentRow.scaling_group == self.name
+            )
+            result = (await db_session.execute(query_stmt)).fetchall()
```

Comment:
`ExtendedAsyncSAEngine.begin_readonly()` returns `Connection` object and it allows SQLAlchemy Core APIs only. 
How about using `ExtendedAsyncSAEngine.begin_readonly_session()` to open SQLAlchemy session and uses ORM APIs?

Reply:
I never notice, thank you for the review! 👍